auth_server/src/realms.rs

```diff
@@ -61,6 +65,39 @@ pub async fn receive_realm_pings(auth_db: std::sync::Arc<AuthDatabase>) -> Resul
     }
 }
 
+async fn get_realm_list(auth_database: std::sync::Arc<AuthDatabase>, account_id: u32) -> Result<Vec<Realm>> {
+    //TODO(wmxd): it will be good idea to cache the database stuff
+    //TODO(wmxd): for now it will be better select realms and number_of_chars in one database trip (eg: left join)
+    let db_realms = auth_database.get_all_realms().await?;
+    let mut realms = Vec::with_capacity(db_realms.len());
+    for realm in db_realms {
+        let num_characters = auth_database.get_num_characters_on_realm(account_id, realm.id).await?;
+
+        // TODO: Use flags from DB.
+        let mut flag = Realm_RealmFlag::empty();
+
+        if realm.online == 0 {
+            flag = flag.set_OFFLINE();
+        }
+
+        let realm_type: RealmType = RealmType::try_from(realm.realm_type).unwrap_or_default();
+
+        realms.push(Realm {
+            realm_type,
+            locked: 0,
+            flag,
+            name: realm.name,
+            address: realm.ip,
+            population: Population::GreenRecommended,
```

Comment:
The Population in your messages/db seems to be an f32 which is population_count / MAX_POPULATION. I'm not sure how to convert that into a u32 for the message, or if the 3.3.5 client is actually meant to receive an f32?

Reply:
```
Gtker — Today at 11:39 AM
It is technically an f32, but wowm doesn't support floats for enums, so I just decided to hardcode the float values as u32
It determines which servers are shown as high/med/low pop, although I don't really know what the algorithm is
```

**Notes from looking at other emulators**

Vmangos:

```c
        float popu = float(GetActiveSessionCount());        // updated number of users on the server
        popu /= pLimit;
        popu *= 2;

        SqlStatement stmt = LoginDatabase.CreateStatement(id, "UPDATE `realmlist` SET `population` = ? WHERE `id` = ?");
```

Always sends a float based on calculation, never hardcoded `Recommended` values.

Reply:
I've changed this logic to convert the f32 into the u32 equivalent representation of the f32, so the client should be able to parse it correctly as an f32 (similar to how the normal enum values are handled).